src/Containers/Microsoft.NET.Build.Containers/Layer.cs

```diff
@@ -163,6 +131,43 @@ string SanitizeContainerPath(string containerPath)
             fs.Position = 0;
 
             SHA256.HashData(fs, hash);
+
+            // Writes a tar entry corresponding to the file system item.
+            static void WriteTarEntryForFile(TarWriter writer, FileSystemInfo file, string containerPath, IEnumerable<KeyValuePair<string, string>> entryAttributes)
+            {
+                UnixFileMode mode = DetermineFileMode(file);
+
+                if (file is FileInfo)
+                {
+                    using var fileStream = File.OpenRead(file.FullName);
+                    PaxTarEntry entry = new(TarEntryType.Directory, containerPath, entryAttributes)
+                    {
+                        Mode = mode,
+                        DataStream = fileStream
+                    };
+                    writer.WriteEntry(entry);
+                }
+                else
+                {
+                    PaxTarEntry entry = new(TarEntryType.RegularFile, containerPath, entryAttributes)
+                    {
+                        Mode = mode
+                    };
+                    writer.WriteEntry(entry);
+                }
+
+                static UnixFileMode DetermineFileMode(FileSystemInfo file)
+                {
+                    const UnixFileMode nonExecuteMode = UnixFileMode.UserRead | UnixFileMode.UserWrite |
+                                                        UnixFileMode.GroupRead |
+                                                        UnixFileMode.OtherRead;
+                    const UnixFileMode executeMode = nonExecuteMode | UnixFileMode.UserExecute | UnixFileMode.GroupExecute | UnixFileMode.OtherExecute;
```

Comment:
When we add a property to control these permissions in https://github.com/dotnet/sdk-container-builds/issues/399.
That property could control `nonExecuteMode`, and we can derive `executeMode` from it by adding x-bit where there are r-bits.

(This matches the note in https://github.com/dotnet/sdk-container-builds/issues/399#issuecomment-1489777710).